src/Domain/Creators/TimeDependence/Composition.tpp

```diff
@@ -0,0 +1,146 @@
+                 "of one the functions of time being composed. This is done as "
+                 "an option to the time dependence in the input file. The "
+                 "corresponding time dependence with the duplicate name is '"
+              << time_dep_names[counter_into_dep_names] << "'.");
+      }
+      functions_of_time.insert(std::move(name_and_func));
+    }
+  };
+  (void)insert_into_map;  // Silence compiler warnings
+  EXPAND_PACK_LEFT_TO_RIGHT(insert_into_map(std::move(functions_of_time_pack)));
+  return functions_of_time;
+}
+}  // namespace detail
+
+template <typename TimeDependenceCompTag0, typename... TimeDependenceCompTags>
+Composition<TimeDependenceCompTag0, TimeDependenceCompTags...>::Composition(
+    tmpl::type_from<TimeDependenceCompTag0> first_time_dep,
+    tmpl::type_from<TimeDependenceCompTags>... rest_time_dep) noexcept
+    : coord_map_(detail::combine_coord_maps(
+          dynamic_cast<typename TimeDependenceCompTag0::time_dependence&>(
+              *first_time_dep)
+              .map_for_composition(),
+          dynamic_cast<typename TimeDependenceCompTags::time_dependence&>(
+              *rest_time_dep)
+              .map_for_composition()...)),
+      functions_of_time_(detail::combine_functions_of_time(
+          {TimeDependenceCompTag0::name(), TimeDependenceCompTags::name()...},
+          std::move(first_time_dep->functions_of_time()),
+          std::move(rest_time_dep->functions_of_time())...)) {}
+
+template <typename TimeDependenceCompTag0, typename... TimeDependenceCompTags>
+auto Composition<TimeDependenceCompTag0, TimeDependenceCompTags...>::get_clone()
+    const noexcept -> std::unique_ptr<TimeDependence<mesh_dim>> {
+  return std::make_unique<Composition>(coord_map_, functions_of_time_);
+}
+
+template <typename TimeDependenceCompTag0, typename... TimeDependenceCompTags>
+auto Composition<TimeDependenceCompTag0, TimeDependenceCompTags...>::block_maps(
+    const size_t number_of_blocks) const noexcept
+    -> std::vector<std::unique_ptr<
+        domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, mesh_dim>>> {
+  std::vector<std::unique_ptr<
+      domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, mesh_dim>>>
+      result{number_of_blocks};
+  result[0] = std::make_unique<CoordMap>(coord_map_);
+  for (size_t i = 1; i < number_of_blocks; ++i) {
+    result[i] = result[0]->get_clone();
+  }
+  return result;
+}
+
+template <typename TimeDependenceCompTag0, typename... TimeDependenceCompTags>
+auto Composition<TimeDependenceCompTag0,
+                 TimeDependenceCompTags...>::functions_of_time() const noexcept
+    -> std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>> {
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      result{};
+  for (const auto& name_and_function_of_time : functions_of_time_) {
```

Comment:
seems like the copy of an `unordered_map` of `unique_ptr` might be done frequently. If you anticipate repeating this same procedure more than the couple of times in this file, consider creating a helper function for it.